includes/Admin/Tools.php

```diff
@@ -420,4 +435,112 @@ class="cp-tools-textarea" name="cp-sysinfo" style="width:100%;height: 70vh"
 		<?php
 	}
 
+	/**
+	 * Exports all messages when form is submitted
+	 * @return void
+	 * @since 1.0.4
+	 * @author Jonathan Roley
+	 */
+	public function export_data() {
+		$return_value = [];
+
+		$args = [
+			'post_type'      =>  CP_LIBRARY_UPREFIX . "_item",
+			'post_status'    => array('publish','private','draft','future'),
+			'posts_per_page' => -1,
+		];
+
+		$posts = new \WP_Query( $args );
+
+		$file_handle = fopen('php://memory', 'w');
```

Comment:
@jjroley Are we at all concerned that this file may exceed a conservatively-rationed server's memory limit? Do we have a rough idea of how large this file might reasonably become?